Modules/System/WebDAV/src/WebDAVClientImpl.Codeunit.al

```diff
@@ -0,0 +1,192 @@
+{
+    Access = Internal;
+
+    var
+        WebDAVOperationResponse: Codeunit "WebDAV Operation Response";
+        Authorization: Interface "WebDAV Authorization";
+        Uri: Text;
+
+    procedure Initialize(InitUri: Text; InitAuthorization: Interface "WebDAV Authorization")
+    begin
+        Uri := InitUri;
+        Authorization := InitAuthorization;
+    end;
+
+    procedure MakeCollection(CollectionName: Text): Boolean
+    var
+        WebDAVRequestHelper: Codeunit "WebDAV Request Helper";
+    begin
+        WebDAVRequestHelper.SetAuthorization(Authorization);
+        WebDAVOperationResponse := WebDAVRequestHelper.MkCol(Uri, CollectionName);
+
+        if not WebDAVOperationResponse.GetDiagnostics().IsSuccessStatusCode() then
+            exit(false);
+
+        exit(true);
+    end;
+
+    procedure Delete(MemberName: Text): Boolean
+    var
+        WebDAVRequestHelper: Codeunit "WebDAV Request Helper";
+    begin
+        WebDAVRequestHelper.SetAuthorization(Authorization);
+        WebDAVOperationResponse := WebDAVRequestHelper.Delete(Uri, MemberName);
+
+        if not WebDAVOperationResponse.GetDiagnostics().IsSuccessStatusCode() then
+            exit(false);
+
+        exit(true);
+    end;
+
+    procedure Put(Content: InStream): Boolean
+    var
+        HttpContent: HttpContent;
+    begin
+        HttpContent.WriteFrom(Content);
+        exit(Put(HttpContent));
+    end;
+
+    procedure Put(Content: Text): Boolean
+    var
+        HttpContent: HttpContent;
+    begin
+        HttpContent.WriteFrom(Content);
+        exit(Put(HttpContent));
+    end;
+
+    procedure Put(Content: HttpContent): Boolean
+    var
+        WebDAVRequestHelper: Codeunit "WebDAV Request Helper";
+    begin
+        WebDAVRequestHelper.SetAuthorization(Authorization);
+        WebDAVOperationResponse := WebDAVRequestHelper.Put(Uri, Content);
+
+        if not WebDAVOperationResponse.GetDiagnostics().IsSuccessStatusCode() then
+            exit(false);
+
+        exit(true);
+    end;
+
+    procedure Move(Destination: Text): Boolean
+    var
+        WebDAVRequestHelper: Codeunit "WebDAV Request Helper";
+    begin
+        WebDAVRequestHelper.SetAuthorization(Authorization);
+        WebDAVOperationResponse := WebDAVRequestHelper.Move(Uri, Destination);
+
+        if not WebDAVOperationResponse.GetDiagnostics().IsSuccessStatusCode() then
+            exit(false);
+
+        exit(true);
+    end;
+
+    procedure Copy(Destination: Text): Boolean
+    var
+        WebDAVRequestHelper: Codeunit "WebDAV Request Helper";
+    begin
+        WebDAVRequestHelper.SetAuthorization(Authorization);
+        WebDAVOperationResponse := WebDAVRequestHelper.Copy(Uri, Destination);
+
+        if not WebDAVOperationResponse.GetDiagnostics().IsSuccessStatusCode() then
+            exit(false);
+
+        exit(true);
+    end;
+
+    procedure GetFilesAndCollections(var WebDAVContent: Record "WebDAV Content"; Recursive: Boolean): Boolean
```

Comment:
All the "WebDAV Content Parser" functions are NonDebuggable, such as the Initialize, if that shouldn't be debuggable, then these functions shouldn't be debuggable either.
Personally I don't see any reason why the Initialize function is NonDebuggable though, so maybe that one should have the tag removed.